### PR TITLE
Smash wall and kill player

### PR DESCRIPTION
**Pull request by redjoey**
Smashing walls and a secret command

### CODE REVIEW

File: invadedhouse/commands.py [hunk 1]
```diff
@@ -24,7 +28,10 @@ def get_next_command() -> str:
         return cmd
 
     def interpret_command(self, cmd: str):
-        if cmd == 'exit':
+        if cmd in ('exit', 'bye', 'i\'m leaving', 'i\'m gonna exit', 'end game', 'quit', 'goodbye', 'leave', 'leave so soon', 'delete my progress', 'get outta here', 'skedaddle',
+        'let me out', 'stop', 'end', 'avengers endgame', 'im scared of this place i wanna go home', 'i wanna go home', 'i want my mommy', 'i want my daddy', 'im out', 'goodbye',
+        'this game is trash', 'i dont like this game', 'i dont like this', 'stop running run.py', f'go from {self.player.get_location()} to my house', 'walk into the strange portal that is on the wall that leads to my home',
+        'when is this going to end', f'create a portal to my house in {self.player.get_location()}', 'i hate you', 'exit the game'):
```
Comment (redjoey) [inline invadedhouse/commands.py:34]: all of the things you can type in instead of "exit"

File: invadedhouse/commands.py [hunk 2]
```diff
@@ -145,5 +152,31 @@ def interpret_command(self, cmd: str):
...
+            place_that_the_hole_leads_to = self.house.find_room_by_name('outside')
+            room_that_lets_you_into_the_sword_room = self.house.find_room_by_name('entry room')
+            room_with_a_wall_that_can_be_smashed = self.house.find_room_by_name('sword room')
+            if not self.player.get_location() == room_with_a_wall_that_can_be_smashed:
+                print('Smash attempt failed.')
+                return
+            else:
+                print('The wall has been smashed! You walk through your hole in the wall and you are outside.')
+                print('')
+                room_with_a_wall_that_can_be_smashed.set_exits([room_that_lets_you_into_the_sword_room, place_that_the_hole_leads_to])
```
Comment (redjoey) [inline invadedhouse/commands.py:165]: Smash a wall!

File: invadedhouse/commands.py [hunk 3]
```diff
@@ -145,5 +152,31 @@ def interpret_command(self, cmd: str):
...
+                self.player.take_damage(self.player.get_hp())
+            elif are_you_sure in('no', 'n'):
+                cprint('I thought so. Continue playing.', 'green')
+                print('')
+                return
+            else:
+                print('That\'s not an option. Continue playing.')
+                print('')
+                return
+
```
Comment (redjoey) [inline invadedhouse/commands.py]: secret command that lets you kill yourself

File: invadedhouse/commands.py [hunk 4]
```diff
@@ -1,8 +1,12 @@
 import sys
+from termcolor import cprint
 
 from models.chest import Chest
 from models.house import House
 from models.player import Player
+from models.room import Room
+from models.item import Item
+from models.weapon import Weapon
```
Comment (redjoey) [inline invadedhouse/commands.py:9]: ehh, just in case

File: invadedhouse/commands.py [hunk 5]
```diff
@@ -1,8 +1,12 @@
 import sys
+from termcolor import cprint
```
Comment (redjoey) [inline invadedhouse/commands.py:2]: we need this for the secret command

File: invadedhouse/models/house.py [hunk 6]
```diff
@@ -69,6 +69,10 @@ def build_house(self):
             'Nobody drew this on the map... Well, you can warp somewhere. Didn\'t see this on'
             ' the map. Just sayin\'.'
         )
+        outside_1 = Room(
+            'Outside',
+            'The sun is out, the ground is grassy, and you are outside.'
+        )
```
Comment (redjoey) [inline invadedhouse/models/house.py:75]: outside is now a room

File: invadedhouse/models/house.py [hunk 7]
```diff
@@ -117,6 +121,11 @@ def build_house(self):
             'Not so awesome. Hey, did you loot the trash can? Let\'s try the sword in there.',
             1
         )
+        knife = Weapon(
+            'Knife',
+            'Dangerous kitchen utencil.',
+            5
+        )
```
Comment (redjoey) [inline invadedhouse/models/house.py:128]: I was originally gonna use this for the secret command, but I didn't. It is a secret weapon, though, so let's keep it

File: invadedhouse/models/house.py [hunk 8]
```diff
@@ -170,6 +179,8 @@ def build_house(self):
         finale_room.set_exits([laser_room, end_room])
         finale_room.set_monsters([gold_monster])
         warp_room.set_exits([boss_room, bedroom, laser_room])
+        outside_1.set_exits([sword_room])
+        outside_1.set_items([dirt, dirt, rock, apple, knife])
```
Comment (redjoey) [inline invadedhouse/models/house.py:183]: set exits and set items for outside

File: invadedhouse/models/house.py [hunk 9]
```diff
@@ -185,7 +196,8 @@ def build_house(self):
             'laser room': laser_room,
             'finale room': finale_room,
             'end room': end_room,
-            'warp room': warp_room
+            'warp room': warp_room,
+            'outside': outside_1
```
Comment (redjoey) [inline invadedhouse/models/house.py:200]: added outside to the room directory

File: invadedhouse/models/player.py [hunk 10]
```diff
@@ -80,4 +80,14 @@ def gets_attacked_by(self, monster: Monster):
...
+    
+    def take_damage(self, damage):
+        self.hp -= damage
+        print(f'You took {damage} damage!')
+        print('')
+        if self.is_dead():
+            cprint('You died!', 'red')
+            sys.exit()
+        else:
+            print(f'You have {self.get_hp()} of {self.get_max_hp()} HP remaining.')
```
Comment (redjoey) [inline invadedhouse/models/player.py:92]: My **SUPER COOL NEW FUNCTION I MADE!** You like it?

File: invadedhouse/run.py [hunk 11]
```diff
@@ -19,10 +19,11 @@
...
 
 def main():
-    cprint('   ____                       __         __  __ __                   ', 'cyan')
-    cprint('  /  _/___   _  __ ___  _ ___/ /___  ___/ / / // /___   __ __ ___  ___', 'cyan')
-    cprint(' _/ / / _ \ | |/ // _  `// _  // -_)/ _  / / _  // _ \ / // /(_-</  -_)', 'cyan')
+    cprint('   ____                       __         __  __ __                   ', 'red')
+    cprint('  /  _/___   _  __ ___  _ ___/ /___  ___/ / / // /___   __ __ ___  ___', 'yellow')
+    cprint(' _/ / / _ \ | |/ // _  `// _  // -_)/ _  / / _  // _ \ / // /(_-</  -_)', 'green')
     cprint('/___//_//_/ |___/ \_,_/ \__,_/ __/ \_,_/  /_//_/ ___/ \__,_//___/ \__/ ', 'cyan')
+    print('')
```
Comment (redjoey) [inline invadedhouse/run.py:26]: Cool rainbow design for logo, and added an empty line like you said @jsoconnell

File: invadedhouse/commands.py (hunk 3 above)
Comment (redjoey) [reply invadedhouse/commands.py]: mwahahahaha

File: invadedhouse/commands.py [hunk 12]
```diff
@@ -145,5 +152,31 @@ def interpret_command(self, cmd: str):
...
+            else:
+                print('The wall has been smashed! You walk through your hole in the wall and you are outside.')
+                print('')
+                room_with_a_wall_that_can_be_smashed.set_exits([room_that_lets_you_into_the_sword_room, place_that_the_hole_leads_to])
+        elif cmd in('stab me', 'stab myself'):
+            cprint('Do you really want to stab yourself?', 'red')
+            are_you_sure = input('> ')
+            if are_you_sure in('yes', 'y'):
+                cprint('Okay...', 'red')
+                self.player.take_damage(self.player.get_hp())
```
Comment (redjoey) [inline invadedhouse/commands.py:171]: my super cool new function in action

Comment (redjoey) [reply invadedhouse/commands.py:171]: `self.player.take_damage(self.player.get_hp())`

Comment (redjoey) [reply invadedhouse/commands.py:171]: in here it makes the player take as much damage as their hp so they will just die

File: invadedhouse/commands.py (hunk 1 above)
Comment (redjoey) [reply invadedhouse/commands.py:34]: my favorite one is `f'create a portal to my house in {self.player.get_location()}'`.

File: invadedhouse/commands.py [hunk 13]
```diff
@@ -145,5 +152,31 @@ def interpret_command(self, cmd: str):
                     print(f'{item_str} is empty - nothing to loot!')
             else:
                 print('Can\'t loot. Try to loot a different loot.')
+        elif cmd == 'smash a wall' or cmd == 'smash the wall':
```
Comment (jsoconnell) [inline invadedhouse/commands.py:155]: you can do `in` here now too

File: invadedhouse/commands.py (hunk 4 above)
Comment (jsoconnell) [reply invadedhouse/commands.py:9]: lol, that's not how imports work haha

File: invadedhouse/commands.py (hunk 12 above)
Comment (jsoconnell) [reply invadedhouse/commands.py:171]: we probably shouldn't let the player commit suicide, but maybe there is another way to accidentally take damage if you do something dangerous... like if you try to eat a weapon or something. we need to keep our game PG rated! 😇

File: invadedhouse/models/player.py [hunk 14]
```diff
@@ -80,4 +80,14 @@ def gets_attacked_by(self, monster: Monster):
             sys.exit()
         else:
             print(f'You have {self.get_hp()} of {self.get_max_hp()} HP remaining.')
+    
+    def take_damage(self, damage):
+        self.hp -= damage
+        print(f'You took {damage} damage!')
+        print('')
```
Comment (jsoconnell) [inline invadedhouse/models/player.py:87]: fyi you can just do `print()`

File: invadedhouse/models/player.py (hunk 10 above)
Comment (jsoconnell) [reply invadedhouse/models/player.py:92]: 👏this function is AWESOME, great job 👏

File: invadedhouse/run.py (hunk 11 above)
Comment (jsoconnell) [reply invadedhouse/run.py:26]: nice nice nice, you got the hang of cprint right away! 🌈

File: invadedhouse/commands.py (hunk 2 above)
Comment (jsoconnell) [reply invadedhouse/commands.py:165]: this was also a super super clever command that you added, you're really learning how to control what the user can do and change the state of the game after they do it!

another thing we _could_ do is something like `Room.add_exit` so you don't have lookup the existing exits just to add a new one... but it is SUPER CLEVER that you figured out how to do that!!

**Review by jsoconnell — CHANGES_REQUESTED** [review]
Your best coding yet, GREAT JOB @redjoey !!

I do want you to change the stab me command though, we don't want to promote that kind of violence in our game... I gave you suggestion on another way to still use your super cool `player.take_damage` function, I really love that one!

File: invadedhouse/commands.py (hunk 12 above)
Comment (redjoey) [reply invadedhouse/commands.py:171]: oh ok got it

Comment (redjoey) [reply invadedhouse/commands.py:171]: okay i fixed the command it doesnt sound evil anymore

File: invadedhouse/models/player.py (hunk 14 above)
Comment (redjoey) [reply invadedhouse/models/player.py:87]: oh!

**Comment by redjoey** [comment]
i fixed the command now so can i merge? or what do i do?

File: invadedhouse/commands.py (hunk 12 above)
Comment (redjoey) [reply invadedhouse/commands.py:171]: fixed

Comment (redjoey) [reply invadedhouse/commands.py:171]: FIXED IT!

Comment (redjoey) [reply invadedhouse/commands.py:171]: umm excuse me?

Comment (redjoey) [reply invadedhouse/commands.py:171]: LOOK I FIXED IT!!!!!!!!!!!!!!!!!!!!!!!!!!!! LOOOOOOOK

Comment (redjoey) [reply invadedhouse/commands.py:171]: look. at. it. please. because. i. fixed. it.

Comment (redjoey) [reply invadedhouse/commands.py:171]: LOOOOOOOOOOOOOOOOOOOK AT IT!

Comment (redjoey) [reply invadedhouse/commands.py:171]: look at it

**Review by jsoconnell — APPROVED** [review]
thanks, that's better 💗

**Pull request merged**